pyccl/ccl_utils.i

```diff
@@ -122,4 +136,48 @@ void get_spline2d_arrays(gsl_spline2d *spline2d,
   memcpy(zarr, spline2d->zarr, sizeof(double)*x_size*y_size);
 }
 
+void get_spline3d_arrays(gsl_spline2d **spline2d,
+                         int x_size, double* xarr,
+                         int y_size, double* yarr,
+                         int t_size, double* tarr,
+                         int na,
+                         int *status)
+{
+  // check for inconsistencies //
+  if(spline2d == NULL) {
+    *status = CCL_ERROR_MEMORY;
+    return;
+  }
+  // check only the first array element; the others must be consistent //
```

Comment:
Maybe add a check for the other elements to the for loop, since the function accepts an array of arbitrary spline pointers, for which the dimensions could differ.

Reply:
Done. I thought I wouldn't loop through every scale factor to save a bit of runtime as this is a very specialized function, but better to be on the safe side.